packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -315,3 +294,64 @@ class ServerApi {
   FutureOr<shelf.Response> notImplemented() =>
       shelf.Response(HttpStatus.noContent);
 }
+
+abstract class _ExtensionsApiHandler {
+  static FutureOr<shelf.Response> handleServeAvailableExtensions(
+    ServerApi api,
+    Map<String, String> queryParams,
+    ExtensionsManager extensionsManager,
+  ) async {
+    final missingRequiredParams = await ServerApi._checkRequiredParameters(
```

Comment:
Something doesn't feel right about this. Do we need to return `FutureOr<T>` for any of the methods in this class? AFAICT, it's just causing async pollution throughout this file. We definitely shouldn't need to do any asynchronous work to check an entry in a map.

`FutureOr<T>` should really only be used when declaring a type for a callback can either be synchronous or asynchronous.

Consider updating the functions in this file to only return `shelf.Response` if they're not doing any asynchronous work. Otherwise, they can return `Future<shelf.Response>` where appropriate.

Reply:
good catch. The `ServerApi.handle` method still needs to return `FutureOr` since some of the api handlers do async activity, but I cleaned up what I could.